### DISCUSSION TITLE
Are there plans to provide .deb, .rpm, ... library packages?

### DISCUSSION THREAD

**Discussion by FlorianSchwendinger** [ERGO-Code/HiGHS#961] [Announcements]
Since `cmake` makes it very easy to build `.deb` + `.rpm` packages I wonder if there are plans to release .deb, .rpm, ... libraries?

**Comment by odow** [comment]
From my point of view (I maintain the current binaries https://github.com/ERGO-Code/HiGHS#download as part of the JuMP project in Julia and am not part of the HiGHS team):

Given the limited developer time of the core HiGHS team, I'd strongly recommend that they focus on HiGHS and leave packaging to the wider community. I don't think there are any technical reasons stopping someone else from doing it?

**Comment by FlorianSchwendinger** [comment]
Yes there no technical reasons nor license reasons, stopping anyone to upload the packaged versions.
However, it would seam beneficial to coordinate first.

Or to put in a different way, I can try to get them into the corresponding repos since this would make my r-package development easier (on the CRAN side I would only be responsible for the code I wrote not also for the code HiGHS). But it would only make sense to do so if there not already plans in place for that and thats what I am asking currently about. :)

**Comment by jajhall** [comment]
We have no plans to provide these packages so you won't be duplicating any activity by developing them yourself. If they are useful for you then they may be for others, so contributing them is most welcome, thanks.

**Reply by DDAK** [reply]
@FlorianSchwendinger I have Dockerized HiGHs and you should be able to run it on any platform that supports docker. You can check here: https://github.com/DDAK/HiGHS. I can do a PR and add it to the main project, if it is interesting.

**Reply by FlorianSchwendinger** [reply]
Thanks, but the goal is to directly link to the library from **R** so docker doesn't help.

**Reply by spakin** [reply]
You can find `.deb` packages of HiGHS on my Ubuntu PPA: https://launchpad.net/~pakin/+archive/ubuntu/other

**Reply by jajhall** [reply]
Thanks @spakin . I see that you've installed v1.4.2. Will you update this when we release new versions?

I guess that this allows easy installation of HiGHS on Linux. Could you indicate how this is done so that we can publicise it?

Thanks!

**Reply by spakin** [reply]
> Thanks @spakin . I see that you've installed v1.4.2. Will you update this when we release new versions?

I hope to but can't promise I'll do this in perpetuity.  If someone closer to the project would like to take over maintenance, I'd be fine with that.  Otherwise, feel free to shoot me emails occasionally to nag me to update the packages if I fall behind.  You can use `scott` dash `highs` at `pakin.org`.

> I guess that this allows easy installation of HiGHS on Linux. Could you indicate how this is done so that we can publicise it?

Sure.  The following is the one-time setup users need to do:

1. Point your Ubuntu distribution to my PPA:
```bash
sudo add-apt-repository ppa:pakin/other
sudo apt update
```
2. Install HiGHS with `sudo apt install highs` (to run the `highs` command) and/or `sudo apt install libhighs-dev` (to program with the HiGHS API).

I'm sure there are GUI equivalents to the above, but I'm a command-line person myself.  Once that setup is complete, HiGHS will be updated along with all other Ubuntu packages installed on the system.

As additional information,  I divided HiGHS into three packages:

* `highs`: the `highs` executable and man page
* `libhighs-dev`: the HiGHS include files, shared object suitable for linking, and examples
* `libhighs1`: the HiGHS shared object needed to run `highs` and any user-developed programs

The first two depend on the third, which rarely needs to be installed explicitly.  (A contrived exception would be if someone gives you a program compiled against the HiGHS API that you want to run, but you want to install as little new software as possible.)

At the time of this writing, I provide packages for Ubuntu Focal (20.04), Jammy (22.04), Kinetic (22.10), and Lunar (23.04).  Older Ubuntu distributions can't easily build HiGHS because HiGHS's build process requires too new of a CMake.

**Reply by jajhall** [reply]
Thanks @spakin I've noted your instructions and will have a go later. At the moment we don't have the resources to take over maintenance.
